File: option_desk/positions/calculations.py
from __future__ import annotations


def calculate_profit_capture(entry_premium: float, mark: float) -> float | None:
    if entry_premium <= 0:
        return None
    return round((entry_premium - mark) / entry_premium, 4)


def calculate_mark_pnl(entry_premium: float, mark: float, contracts: int) -> float:
    return round((entry_premium - mark) * contracts * 100, 2)


def calculate_estimated_close_pnl(entry_premium: float, close_price: float, contracts: int) -> float:
    return round((entry_premium - close_price) * contracts * 100, 2)

File: web/src/positions/calculations.ts
export function roundMoney(value: number): number {
  return Math.round(value * 100) / 100;
}

export function realizedOptionPnl(
  entryPremium: number,
  closePrice: number,
  contracts: number,
  fees = 0,
): number {
  return roundMoney((entryPremium - closePrice) * contracts * 100 - fees);
}

export function markOptionPnl(entryPremium: number, mark: number, contracts: number): number {
  return roundMoney((entryPremium - mark) * contracts * 100);
}

export function profitCapture(entryPremium: number, mark: number): number | null {
  if (entryPremium <= 0) return null;
  return (entryPremium - mark) / entryPremium;
}
